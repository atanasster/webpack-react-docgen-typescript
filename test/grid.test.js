import compiler from './compiler.js';

jest.setTimeout(100000);

test('grid', async () => {
  const stats = await compiler('./examples/Grid.tsx');
  const jsonStats = stats.toJson();
  const output = jsonStats.modules[1].source;
  expect(output).toMatchSnapshot();
});
