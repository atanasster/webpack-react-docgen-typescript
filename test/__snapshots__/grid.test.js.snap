// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grid 1`] = `
"var _jsxFileName = \\"/Users/atanasster/webpack-react-docgen-typescript/test/examples/Grid.tsx\\";
import * as React from 'react';
/**
 * Grid properties.
 */

/**
 * Form Grid.
 */
export var Grid = function Grid(props) {
  var smaller = function smaller() {
    return;
  };

  return React.createElement(\\"div\\", {
    __source: {
      fileName: _jsxFileName,
      lineNumber: 24
    },
    __self: this
  }, \\"Grid\\");
};

try {
  Grid.__docgenInfo = {
    \\"description\\": \\"Form Grid.\\",
    \\"displayName\\": \\"Grid\\",
    \\"methods\\": [],
    \\"props\\": {
      \\"prop1\\": {
        \\"defaultValue\\": null,
        \\"description\\": \\"prop1 description\\",
        \\"name\\": \\"prop1\\",
        \\"parent\\": {
          \\"fileName\\": \\"webpack-react-docgen-typescript/test/examples/Grid.tsx\\",
          \\"name\\": \\"IGridProps\\"
        },
        \\"required\\": false,
        \\"type\\": {
          \\"name\\": \\"string\\"
        }
      },
      \\"prop2\\": {
        \\"defaultValue\\": null,
        \\"description\\": \\"prop2 description\\",
        \\"name\\": \\"prop2\\",
        \\"parent\\": {
          \\"fileName\\": \\"webpack-react-docgen-typescript/test/examples/Grid.tsx\\",
          \\"name\\": \\"IGridProps\\"
        },
        \\"required\\": true,
        \\"type\\": {
          \\"name\\": \\"number\\"
        }
      },
      \\"prop3\\": {
        \\"defaultValue\\": null,
        \\"description\\": \\"prop3 description\\",
        \\"name\\": \\"prop3\\",
        \\"parent\\": {
          \\"fileName\\": \\"webpack-react-docgen-typescript/test/examples/Grid.tsx\\",
          \\"name\\": \\"IGridProps\\"
        },
        \\"required\\": true,
        \\"type\\": {
          \\"name\\": \\"() => void\\"
        }
      }
    },
    \\"type\\": \\"tsType\\"
  };
} catch (e) {//eat exception
}"
`;
